Serialization for nn.quantized.functional modules

raghuramank100 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#24924 Serialization for nn.quantized.functional modules**
* #24923 Support observer without any data calibration
* #24804 Update mapping dictionary to support functionalmodules and pooling operations
* #24803 Handle empty qconfig for functional Modules
* #24789 Work around for bias quantization for conv and linear operators

Add load_from_state_dict and save_from_state_dict for quantized functional modules

Differential Revision: [D16923651](https://our.internmc.facebook.com/intern/diff/D16923651/)

raghuramank10000 committed:
Serialization for nn.quantized.functional modules

Add load_from_state_dict and save_from_state_dict for quantized functional modules

Differential Revision: [D16923651](https://our.internmc.facebook.com/intern/diff/D16923651/)

raghuramank10000 committed:
Update on "Serialization for nn.quantized.functional modules"


Add load_from_state_dict and save_from_state_dict for quantized functional modules

Differential Revision: [D16923651](https://our.internmc.facebook.com/intern/diff/D16923651/)

raghuramank10000 committed:
Update on "Serialization for nn.quantized.functional modules"


Add load_from_state_dict and save_from_state_dict for quantized functional modules

Differential Revision: [D16923651](https://our.internmc.facebook.com/intern/diff/D16923651/)

jerryzh168 commented on torch/nn/quantized/modules/functional_modules.py:
```diff
@@ -73,6 +73,24 @@ class QFunctional(torch.nn.Module):
+
+    def _save_to_state_dict(self, destination, prefix, keep_vars):
+        super(QFunctional, self)._save_to_state_dict(destination, prefix, keep_vars)
+        destination[prefix + 'scale'] = torch.tensor(self.scale)
```
do we need to special case scale and zero_point?

raghuramank10000 committed:
Update on "Serialization for nn.quantized.functional modules"


Add load_from_state_dict and save_from_state_dict for quantized functional modules

Differential Revision: [D16923651](https://our.internmc.facebook.com/intern/diff/D16923651/)

jerryzh168 reviewed (approved):
LGTM

raghuramank10000 committed:
Update on "Serialization for nn.quantized.functional modules"


Add load_from_state_dict and save_from_state_dict for quantized functional modules

Differential Revision: [D16923651](https://our.internmc.facebook.com/intern/diff/D16923651/)

raghuramank10000 committed:
Update on "Serialization for nn.quantized.functional modules"


Add load_from_state_dict and save_from_state_dict for quantized functional modules

Differential Revision: [D16923651](https://our.internmc.facebook.com/intern/diff/D16923651/)

raghuramank10000 committed:
Update on "Serialization for nn.quantized.functional modules"


Add load_from_state_dict and save_from_state_dict for quantized functional modules

Differential Revision: [D16923651](https://our.internmc.facebook.com/intern/diff/D16923651/)
